packages/migrate/src/__tests__/MigrateDiff.test.ts

```diff
@@ -0,0 +1,513 @@
+    })
+
+    it('should fail if missing --from-... and --to-...', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              0 \`--from-...\` paramater(s) provided. 1 must be provided.
+              0 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if only --from-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--from-empty'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              0 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if only --to-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--to-empty'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              0 \`--from-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if more than 1 --from-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--from-empty', '--from-url=file:dev.db'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              2 \`--from-...\` paramater(s) provided. 1 must be provided.
+              0 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if more than 1 --to-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--to-empty', '--to-url=file:dev.db'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              0 \`--from-...\` paramater(s) provided. 1 must be provided.
+              2 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if schema does no exists, --from-schema-datasource', async () => {
+      ctx.fixture('empty')
+      expect.assertions(2)
+
+      try {
+        await MigrateDiff.new().parse([
+          '--preview-feature',
+          '--from-schema-datasource=./doesnoexists.prisma',
+          '--to-empty',
+        ])
+      } catch (e) {
+        expect(e.code).toEqual(undefined)
+        expect(e.message).toMatchInlineSnapshot(`
+          Could not find a schema.prisma file that is required for this command.
+          You can either provide it with --schema, set it as \`prisma.schema\` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
```

Comment:
Does this work with the client as well?

Reply:
What do you mean?

Note for myself: the error message is confusing here, to check and fix

Reply:
To have a 'prisma.schema' in the package.json

Reply:
Yes, for example, generate calls `getSchemaPathInternal` which 
```
  // 1. Try the user custom path
  // 2. Try the package.json `prisma.schema` custom path
  // 3. Try the conventional `./schema.prisma` or `./prisma/schema.prisma` paths
  // 4. Try resolving yarn workspaces and looking for a schema.prisma file there
```
See https://github.com/prisma/prisma/blob/main/packages/sdk/src/cli/getSchema.ts#L284-L295

Reply:
TIL :100: